great_expectations/experimental/datasources/sources.py

```diff
@@ -122,25 +122,27 @@ def _register_assets(cls, ds_type: Type[Datasource], asset_type_lookup: TypeLook
                 f"No `{ds_type.__name__}.asset_types` have be declared for the `Datasource`"
             )
 
-        for t in asset_types:
-            if t.__name__.startswith("_"):
+        for asset_type in asset_types:
+            if asset_type.__name__.startswith("_"):
                 logger.debug(
-                    f"{t} is private, assuming not intended as a public concrete type. Skipping registration"
+                    f"{asset_type} is private, assuming not intended as a public concrete type. Skipping registration"
                 )
                 continue
             try:
-                asset_type_name = t.__fields__["type"].default
+                asset_type_name = asset_type.__fields__["type"].default
                 if asset_type_name is None:
                     raise TypeError(
-                        f"{t.__name__} `type` field must be assigned and cannot be `None`"
+                        f"{asset_type.__name__} `type` field must be assigned and cannot be `None`"
                     )
                 logger.debug(
-                    f"2b. Registering `DataAsset` `{t.__name__}` as {asset_type_name}"
+                    f"2b. Registering `DataAsset` `{asset_type.__name__}` as {asset_type_name}"
                 )
-                asset_type_lookup[t] = asset_type_name
+                asset_type_lookup[
+                    (ds_type, asset_type)
+                ] = f"{ds_type.__name__}.{asset_type_name}"
```

Comment:
I would solve this a different way, by special casing the `in-memory` `PandasDatsource` asset names rather than changing the format for every ZEP asset type.